app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,26 @@
+class SessionsController < ApplicationController
+
+  def new
+    @user = User.new
+  end
+
+  def create
+    @user = User.find_by user_name: params[:user][:user_name]
+
+    if @user
+      session[:user_id] = @user.id
+      flash[:success] = "Welcome back #{@user.user_name}"
+    else
+      @user = User.create user_name: params[:user][:user_name]
```

Comment:
You need to check to verify that `create` worked!  I tried to log in with a blank username and got a "welcome" message!  :p